homeassistant/components/qnap_qsw/config_flow.py

```diff
@@ -63,3 +70,61 @@ async def async_step_user(
             ),
             errors=errors,
         )
+
+    async def async_step_dhcp(self, discovery_info: dhcp.DhcpServiceInfo) -> FlowResult:
+        """Handle DHCP discovery."""
+        self._discovered_url = f"http://{discovery_info.ip}"
+        self._discovered_mac = discovery_info.macaddress
+
+        _LOGGER.debug("DHCP discovery detected QSW: %s", self._discovered_mac)
+
+        mac = format_mac(self._discovered_mac)
+        options = ConnectionOptions(self._discovered_url, "", "")
+        qsw = QnapQswApi(aiohttp_client.async_get_clientsession(self.hass), options)
+
+        try:
+            await qsw.get_live()
+        except QswError as err:
+            raise AbortFlow("cannot_connect") from err
+
+        await self.async_set_unique_id(format_mac(mac))
+        self._abort_if_unique_id_configured()
+
+        return await self.async_step_discovered_connection(None)
```

Comment:
```suggestion
        return await self.async_step_discovered_connection()
```

`None` is the default